server/src/test/java/org/elasticsearch/search/fetch/subphase/FieldFetcherTests.java

```diff
@@ -394,21 +397,288 @@ public void testTextSubFields() throws IOException {
         }
     }
 
-    private static Map<String, DocumentField> fetchFields(MapperService mapperService, XContentBuilder source, String fieldPattern)
-        throws IOException {
+    public void testSimpleUnmappedFields() throws IOException {
```

Comment:
Nice test coverage! Some other cases that'd be good to check:
* Source documents that have dots in field names (like `unmapped_field.a: value`).
* Mixed dot and object notation. We recently found a bug in the 'fields' option related to this: https://github.com/elastic/elasticsearch/issues/65499.

Reply:
Adding tests here...